crates/configuration/src/configuration.rs

```diff
@@ -79,9 +80,9 @@ pub async fn parse_configuration(
         }?;
     Ok(Configuration {
         metadata: configuration.metadata,
-        pool_settings: configuration.pool_settings,
+        pool_settings: configuration.connection_settings.pool_settings,
         connection_uri,
-        isolation_level: configuration.isolation_level,
-        mutations_version: configuration.configure_options.mutations_version,
+        isolation_level: configuration.connection_settings.isolation_level,
+        mutations_version: configuration.mutations_version,
```

Comment:
Perhaps it's worth just storing `connection_settings` directly in the `Configuration`.

Reply:
Thought about it, but didn't want to make too many changes, and wasn't sure about whether we should use version3 structure here. Maybe we can decide in a separate PR since this is internal to the connector?

Reply:
In general I do agree that that's a better structure though.